Handle Enter/Space for additional a11y roles

comp615 opened the pull request:
Addresses #1068, there may be more roles to handle as well, but they can be added as they come up?

necolas commented on packages/react-native-web/src/exports/createElement/index.js:
```diff
@@ -23,6 +23,8 @@ EventPluginHub.injection.injectEventPluginsByName({
 const isModifiedEvent = event =>
   !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
 
+const buttonLikeRoles = ['button', 'menuitem'];
```
Let's use an object rather than an array

necolas commented:
tabindex/focus stuff to check over is here: https://github.com/necolas/react-native-web/blob/e8f2c987864562bda11bec6611db98b7239688d6/packages/react-native-web/src/modules/createDOMProps/index.js#L116

comp615 commented:
Thank for that pointer. Sorry for the delay, we had to upgrade to the new accessibility stuff which was its own task. 

Updated list to accessibility util and moved into its own file for easy reuse. Not sure if that's overkill or not, but just let me know what your preference is!

One question on styles, I saw in createDOMProps `role===button` also applies some reset styles. Should that be changed to include this as well? Or are those intended specifically for role button.

https://github.com/necolas/react-native-web/blob/0e81c6ef2758d4ca9b2099b1d04a4f8c417f0f43/packages/react-native-web/src/modules/createDOMProps/index.js#L153

comp615 commented:
@necolas any thoughts on the update? Not sure how to update it from changes requested status :( maybe it just stays like that forever.
